src/@core/src/genre/domain/entities/genre.ts

```diff
@@ -55,26 +56,43 @@ export class Genre extends AggregateRoot<
     this.name = name;
   }
 
-  addCategoryId(categoryId: CategoryId): void {
-    this.categories_id.set(categoryId.value, categoryId);
+  addCategoryId(categoryId: CategoryId) {
+    const categoriesId = cloneDeep(this.props.categories_id);
+    categoriesId.set(categoryId.value, categoryId);
+
+    Genre.validate({
+      ...this.props,
+      categories_id: categoriesId,
+    });
+
+    this.categories_id = categoriesId;
   }
 
-  removeCategoryId(categoryId: CategoryId): void {
-    this.categories_id.delete(categoryId.value);
+  removeCategoryId(categoryId: CategoryId) {
+    const categoriesId = cloneDeep(this.props.categories_id);
+    categoriesId.delete(categoryId.value);
+
+    Genre.validate({
+      ...this.props,
+      categories_id: categoriesId,
+    });
+
+    this.categories_id = categoriesId;
```

Comment:
Podemos aplicar o mesmo conceito do `addCategoryId` também no `removeCategoryId`.

Reply:
Precisamos do valor validado para atribuir à propriedade.
```suggestion
    removeCategoryId(categoryId: CategoryId) {
    const categoriesId = new Map(this.categories_id);
    categoriesId.delete(categoryId.value);

    Genre.validate({
      ...this.props,
      categories_id: categoriesId,
    });

    this.categories_id = categoriesId;
  }
```